### PR TITLE
Fixed possible crash after storage service failed to start.

### PR DESCRIPTION
**Pull request by monadbobo**

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Good job, An additional question，What's kind of sign type occur?  Why `setupSignalHandler` can't catch it ?

File: src/daemons/StorageDaemon.cpp [hunk 1]
```diff
@@ -109,6 +109,7 @@ int main(int argc, char *argv[]) {
                                                                       paths);
     if (!gStorageServer->start()) {
         LOG(ERROR) << "Storage server start failed";
+        gStorageServer->stop();
```
Comment (dangleptr) [inline src/daemons/StorageDaemon.cpp]: Well done.  I want to know why the daemon crash?

Comment (monadbobo) [reply src/daemons/StorageDaemon.cpp]: You can read the [documentation](https://en.cppreference.com/w/cpp/thread/thread/~thread) for the thread destructor.

> If *this has an associated thread (joinable() == true), std::terminate() is called.

Comment (dangleptr) [reply src/daemons/StorageDaemon.cpp]: which thread?

Comment (monadbobo) [reply src/daemons/StorageDaemon.cpp]: > nebula::WebService::wsThread_.

image

Comment (dangleptr) [reply src/daemons/StorageDaemon.cpp]: I see.  Good catch. 
But i think a better way to solve the problem is to refactor web service.  It has lots of static members inside.  It is hard to handle the lifecycle.  Maybe we should remove them. WDYT?

Of course, we could open another pr for it.

**Comment by dangleptr** [comment]
> Good job, An additional question，What's kind of sign type occur? Why `setupSignalHandler` can't catch it ?

Good question. I want to know it too.

**Comment by dangleptr** [comment]
> > Good job, An additional question，What's kind of sign type occur? Why `setupSignalHandler` can't catch it ?
> 
> Good question. I want to figure it out too.
I tried it locally,  std::terminate() can't be catched by signal(),  we should use std::set_terminate instead.   @bright-starry-sky

**Comment by bright-starry-sky** [comment]
> > Good job, An additional question，What's kind of sign type occur? Why `setupSignalHandler` can't catch it ?
> 
> Good question. I want to figure it out too.
> I tried it locally, std::terminate() can't be catched by signal(), we should use std::set_terminate instead. @bright-starry-sky

Wow, The truth is out！Thanks ! So the terminate should be strengthened.

**Pull request merged**